libopae/plugins/xfpga/metrics/metrics_utils.c

```diff
@@ -870,18 +882,27 @@ fpga_result enum_fpga_metrics(fpga_handle handle)
 		case FPGA_HW_DCP_DC:
 		case FPGA_HW_DCP_VC: {
 
-			// Max10 Power & Thermal
-			result = enum_max10_metrics_info(_handle,
+			memset_s(metrics_path, SYSFS_PATH_MAX, 0);
+			if (sysfs_get_max10_path(_token, metrics_path) == FPGA_OK) {
+
+				// Max10 Power & Thermal
+				result = enum_max10_metrics_info(_handle,
 					&(_handle->fpga_enum_metric_vector),
 					&metric_num,
 					hw_type);
-			if (result != FPGA_OK) {
-				FPGA_MSG("Failed to Enum Power and Thermal metrics.");
+				if (result != FPGA_OK) {
+					FPGA_MSG("Failed to Enum Power and Thermal metrics.");
+				}
 			}
-			// Perf Counters
-			result = enum_perf_counter_metrics(&(_handle->fpga_enum_metric_vector), &metric_num, _token->sysfspath, hw_type);
-			if (result != FPGA_OK) {
-				FPGA_MSG("Failed to Enum Performance metrics.");
+
+			memset_s(metrics_path, SYSFS_PATH_MAX, 0);
+			if (sysfs_get_fme_perf_path(_token, metrics_path) == FPGA_OK) {
+
+				// Perf Counters
+				result = enum_perf_counter_metrics(&(_handle->fpga_enum_metric_vector), &metric_num, _token->sysfspath, hw_type);
+				if (result != FPGA_OK) {
+					FPGA_MSG("Failed to Enum Performance metrics.");
```

Comment:
Seems like an error condition.  Use FPGA_ERR and return an error code?

Reply:
fixed